src/app/models/jobs-response.model.ts

```diff
@@ -1,33 +1,18 @@
-export class JobsResponse {
-  readonly flowStatus: string;
-  readonly flowStatusMessage: string;
-  readonly apiResponse: any;
-  readonly jobs: JobData[];
+import { ApiResponseModel } from "./api-response.model";
+
+export class JobsResponse extends ApiResponseModel {
+  public jobs: JobData[];
 
   constructor(response: any) {
-    try {
-      if (response.flowStatus === 'SUCCESS' || response.flowStatus === 'FAILURE') {
-        this.flowStatus = response.flowStatus;
-        this.flowStatusMessage = response.flowStatusMessage;
-      }
-      else {
-        this.flowStatus = 'FAILURE';
-        this.flowStatusMessage = 'Unable to call API';
-      }
-      if (response.flowStatus === 'SUCCESS') {
-        // save data from response
-        this.apiResponse = response;
-        this.jobs = response.jobs;
-        for (let i = 0; i < this.jobs.length; i++) { // for client-side calculations and data generation for display improvements
-          const jobType = this.jobs[i].jobType;
-          this.jobs[i].prettyPhone = this.prettifyPhone(this.jobs[i].phone);
-          this.jobs[i].jobTypeColor = this.parseJobTypeColor(jobType);
-        }
-      }
-    }
-    catch (e) {
-      this.flowStatus = 'FAILURE';
-      this.flowStatusMessage = 'Unable to parse API response';
+    super(response);
+  }
+
+  protected processResponse(response: any): void {
+    this.jobs = response.jobs;
+    for (let i = 0; i < this.jobs.length; i++) { // for client-side calculations and data generation for display improvements
```

Comment:
Let's add an `Array.isArray()` prior to looping.  We want to verify this is an array.